scipy/stats/_binned_statistic.py

```diff
@@ -650,6 +643,27 @@ def binned_statistic_dd(sample, values, statistic='mean',
     return BinnedStatisticddResult(result, edges, binnumbers)
 
 
+def _calc_binned_statistic(Vdim, binnumbers, result, values, stat_func):
+    unique_bin_numbers = np.unique(binnumbers)
+    for vv in builtins.range(Vdim):
+        bin_map = _create_bin_map(binnumbers, unique_bin_numbers, values, vv)
+        for i in unique_bin_numbers:
+            result[vv, i] = stat_func(bin_map[i])
+
+
+def _create_bin_map(binnumbers, unique_bin_numbers, values, vv):
```

Comment:
same comment as above about signature weirdness w/ `binnumbers` and `unique_bin_numbers`

Reply:
Same comment above. Should we only pass `binnumbers', and the `unique` function should be called inside function? But It calls 'unique' functions multiple times...

Reply:
The thing that I'm unclear about (or was until re-reading tonight) is why the signature needs both `binnumbers` and `unique_bin_numbers`. The point is to construct a hashmap from unique binnumber to binnumbers, I still think the naming of arguments is/are confusing though.